src/main/java/org/candlepin/audit/Event.java

```diff
@@ -104,15 +105,16 @@ public enum ReferenceType {
     // Use with reference id
     private ReferenceType referenceType;
 
-    private String eventData;
+    private Map<String, Object> eventData;
 
     private String messageText;
 
     public Event() {
     }
 
-    public Event(Type type, Target target, String targetName, Principal principal, String ownerId,
-        String consumerUuid, String entityId, String eventData, String referenceId,
+    public Event(Type type, Target target, String targetName,
```

Comment:
From Sonarcloud: this constructor is absurd. Since we're in this area, let's take the time to remove it entirely in favor of chainable mutators.

Reply:
Note that this will require updating the mutators to return the Event object, and we may still need a constructor that takes some immutable fields (if those exist). Ideally we can drop to just a default and maybe one non-default constructor if our hand is forced.

Reply:
I had to keep the default constructor with no parameters for Jackson serialization. However, I added one non-default constructor with parameters for the non-null required fields